websites/E/e621/presence.ts

```diff
@@ -16,95 +16,83 @@ presence.on("UpdateData", async () => {
-			) {
-				presenceData.details = "Searching Posts For:";
+			mTitle = document
+				.querySelector<HTMLMetaElement>('meta[name="og:title"]')
+				?.content.replace(" - e926", "")
+				.replace(" ", "_");
+			document.querySelector<HTMLMetaElement>('meta[name="og:title"]');
+			if (path.includes("posts/")) {
+				mTitle = document.querySelector<HTMLMetaElement>(
+					'meta[property="og:title"]'
+				).content;
+				presenceData.details = "Viewing Post";
+				presenceData.state = `Created By: ${
+					mTitle.slice(0, mTitle.length - 6).split("created by")[1]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View Post",
+						url: document.location.href
+					}
+				];
+			} else if (search.value && search.value !== mTitle) {
+				presenceData.smallImageKey = "search";
+				presenceData.buttons = [
+					{
+						label: "View Search",
+						url: document.location.href
+					}
+				];
+				presenceData.details = "Searching Posts for:";
 				presenceData.state = search.value;
 			} else if (document.location.href.includes("/posts?tags=")) {
-				mTitle = document.querySelector('meta[name="og:title"]');
 				presenceData.details = "Viewing posts about:";
-				presenceData.state = mTitle.content.slice(0, mTitle.content.length - 6);
-			} else if (path.includes("posts/")) {
-				mTitle = document.querySelector('meta[property="og:title"]');
-				const split = mTitle.content
-					.slice(0, mTitle.content.length - 6)
-					.split("created by");
-				presenceData.details = split[0];
-				presenceData.state = `Created By: ${split[1]}`;
+				presenceData.state = mTitle;
 			} else {
+				presenceData.buttons = [
+					{
+						label: "View All Posts",
+						url: document.location.href
+					}
+				];
 				presenceData.details = "All Posts";
 				delete presenceData.state;
 			}
-			if (buttons) {
-				if (
-					(search.value &&
-						search.value !==
-							document.querySelector<HTMLInputElement>("#tags_query").value) ||
-					document.location.href.includes("/posts?tags=")
-				) {
-					presenceData.buttons = [
-						{
-							label: "View Search",
-							url: document.location.href
-						}
-					];
-				} else if (path.includes("posts/")) {
-					presenceData.buttons = [
-						{
-							label: "View Post",
-							url: document.location.href
-						}
-					];
-				} else {
-					presenceData.buttons = [
-						{
-							label: "View All Posts",
-							url: document.location.href
-						}
-					];
-				}
-			}
 		} else if (path.includes("comments")) {
 			title = document.querySelector(
 				"#a-index > div.paginator > menu > li.current-page"
 			);
 			presenceData.details = "Comments";
 			presenceData.state = `Page ${title.textContent}`;
-			if (buttons) {
-				presenceData.buttons = [
-					{
-						label: `View Comment Page ${title.textContent}`,
-						url: document.location.href
-					}
-				];
-			}
+			presenceData.buttons = [
+				{
+					label: `View Comment Page ${title.textContent}`,
```

Comment:
buttons should not include dynamic data
```suggestion
					label: "View Comment Page",
```